pkg/validations/cluster.go

```diff
@@ -9,9 +9,30 @@ import (
 
 func ValidateTaintsSupport(clusterSpec *cluster.Spec) error {
 	if !features.IsActive(features.TaintsSupport()) {
-		if len(clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints) > 0 {
+		wngcTaintsPresent := false
+		for _, wngc := range clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations {
+			if len(wngc.Taints) > 0 {
+				wngcTaintsPresent = true
+				break
+			}
+		}
+
+		if len(clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints) > 0 ||
+			wngcTaintsPresent {
 			return fmt.Errorf("Taints feature is not enabled. Please set the env variable TAINTS_SUPPORT.")
 		}
+	} else if len(clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations[0].Taints) > 0 {
```

Comment:
I think we need to change/remove this. It doesn't really work with multiple worker node groups

Also, this should probably have unit tests, but that could be in a separate ticket

Reply:
https://github.com/aws/eks-anywhere/issues/1126